### PR TITLE
Refactored Dockerfile, and updates to authelia to bring it up to date

### PR DESCRIPTION
**Pull request by codefriar**
Updates the dockerfile creation - much faster for code-only changes. 

Updates Authelia kernel version gating logic to handle kernel versions 5.5+

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -93,59 +38,96 @@ ENV PYTHONPATH $PYTHONPATH
...
     python3-dev \
+    lsb-release \
     python3-distutils \
     python3-wheel \
     python3-venv \
     openssh-server=1:8.4p1-5 \
     libncurses-dev \
-    tmux
+    tmux \
+    wget
```
Comment (voryzen) [inline docker/Dockerfile:65]: wget is already above.. line 103 (prev), line 51 (with changes), it reads 'wget=1.21-1+deb11u1 \'

File: docker/Dockerfile [hunk 2]
```diff
@@ -93,59 +38,96 @@ ENV PYTHONPATH $PYTHONPATH
...
+    apt-get -y install terraform
+###
+# LAYER 3 - Python Requirements
+###
+RUN set -eux && \
+    # Install Python requirements - starting with pip
     pip install --upgrade pip && \
-    pip install wheel && \
-    pip install "https://github.com/VivumLab/py_cui/archive/v0.1.4_beta.tar.gz#egg=py-cui" && \
-    PARAMIKO_REPLACE=1 pip install "https://github.com/ploxiln/paramiko-ng/archive/2.8.10.tar.gz#egg=paramiko" && \
```
Comment (voryzen) [inline docker/Dockerfile:136]: Paramiko is required here; paramiko has issues with ssh keys (can't remember exactly), the issue is fixed with paramiko-ng. Paramiko-ng requires the use of the python egg, so that fabric can use it properly, and the egg doesn't work properly when installed via the requirements file. I just kept the paramiko line in the requirements file, to remind me

TL;DR, this line needs to stay there, otherwise paramiko and fabric don't work

File: docker/Dockerfile [hunk 3]
```diff
@@ -93,59 +38,96 @@ ENV PYTHONPATH $PYTHONPATH
...
+    # Install Hashicorp's terraform
+    apt-get -y install terraform
+###
+# LAYER 3 - Python Requirements
+###
+RUN set -eux && \
+    # Install Python requirements - starting with pip
     pip install --upgrade pip && \
-    pip install wheel && \
-    pip install "https://github.com/VivumLab/py_cui/archive/v0.1.4_beta.tar.gz#egg=py-cui" && \
```
Comment (voryzen) [inline docker/Dockerfile:135]: the py-cui install requirements setting a new python egg, just like paramiko does, so this line will have to stay too

File: requirements.txt [hunk 4]
```diff
@@ -18,3 +18,5 @@ ansible == 5.1.0
 mitogen >= 0.3.2
 jinja2 == 3.0.3
 s3transfer == 0.3.4
+py-cui == 0.1.4
```
Comment (voryzen) [inline requirements.txt:21]: remove this line; the original py-cui does not have the fixes I have added

File: requirements.txt [hunk 5]
```diff
@@ -18,3 +18,5 @@ ansible == 5.1.0
...
+paramiko == 2.11.0
```
Comment (voryzen) [inline requirements.txt:22]: remove this line, we use paramiko-ng, until paramiko sorts the key issue out

File: roles/authelia/tasks/main.yml [hunk 6]
```diff
@@ -13,11 +13,11 @@
       "5.6": "5.6"
       "5.7": "5.7"
       "5.8": "5.8"
-
+      
 - name: Ensure Host Kernel version is adequate for Authelia install
   fail:
     msg: "Your kernel version {{ ansible_kernel }} Does not meet the minimum requirements for Authelia. Please see https://github.com/pimlie/ubuntu-mainline-kernel.sh for a kernel upgrade script"
-  when: "kernel_version is not version(kernel_options[kernel_major_minor], '>=', true)"
+  when: "kernel_version is not version(5.5), '>=', true)" && "kernel_version is not version(kernel_options[kernel_major_minor], '>=', true)"
```
Comment (voryzen) [inline roles/authelia/tasks/main.yml:20]: pre-commit doesn't like this line..



Check failure on line 20 in roles/authelia/tasks/main.yml
GitHub Actions / pre-commit

roles/authelia/tasks/main.yml#L20

syntax error: expected alphabetic or numeric character, but found '&' (syntax)

**Review by voryzen — CHANGES_REQUESTED** [review]
Couple of issues; nice work

**Comment by voryzen** [comment]
Cleaned up all the dependabot PRs, soz about the rebase

File: requirements.txt (hunk 5 above)
Comment (voryzen) [reply requirements.txt]: Fixed

File: docker/Dockerfile (hunk 3 above)
Comment (voryzen) [reply docker/Dockerfile:135]: Fixed

File: docker/Dockerfile (hunk 2 above)
Comment (voryzen) [reply docker/Dockerfile:136]: Fixed

File: docker/Dockerfile (hunk 1 above)
Comment (voryzen) [reply docker/Dockerfile]: Fixed

File: roles/authelia/tasks/main.yml (hunk 6 above)
Comment (voryzen) [reply roles/authelia/tasks/main.yml:20]: @codefriar when removing the ')' after both true statements, pre-commit agrees..
eg. OLD
when: "kernel_version is not version(5.5), '>=', true)" && "kernel_version is not version(kernel_options[kernel_major_minor], '>=', true)"

eg. NEW
when: "kernel_version is not version(5.5), '>=', true" && "kernel_version is not version(kernel_options[kernel_major_minor], '>=', true"

**Comment by voryzen** [comment]
Pushed the little niggling changes. Added a suggestion for the remaining pre-commit error
